test/unit/DCAStrategies/dca-strategies-permissions-handler.spec.ts

```diff
@@ -103,6 +104,52 @@ contract('DCAStrategiesPermissionsHandler', () => {
     });
   });
 
+  describe('set permissions', () => {
+    let tokenId: number = 1;
+    const OPERATOR = constants.NOT_ZERO_ADDRESS;
+    let owner: Wallet;
+    let blockNumber: BigNumber;
+    given(async () => {
+      owner = await wallet.generateRandom();
+      await DCAStrategiesPermissionsHandlerMock.mint(owner.address, [{ operator: OPERATOR, permissions: [] }]);
+      await DCAStrategiesPermissionsHandlerMock.setPermissions(tokenId, [
+        { operator: OPERATOR, permissions: [Permission.INCREASE, Permission.REDUCE] },
+      ]);
+      blockNumber = await DCAStrategiesPermissionsHandlerMock.getBlockNumber();
```

Comment:
Hardhat can be kind of annoying with this, in the sense that between the `setPermissions` call and `getBlockNumber`, the block number could change. I think it would be better if you took a random block number, and set it. That way, you can make sure that it works as expected